lib/tesla/middleware/follow_redirects.ex

```diff
@@ -26,7 +26,9 @@ defmodule Tesla.Middleware.FollowRedirects do
   def call(env, next, opts \\ []) do
     max = Keyword.get(opts || [], :max_redirects, @max_redirects)
 
-    redirect(env, next, max)
+    headers = Enum.filter(env.headers, &(not match?({"Authorization", _}, &1)))
```

Comment:
All tesla adapters are downcasing the header key, so I believe this should check `authorization` not `Authorization`.

However, see the discussion at #362 for the scope.